Add check_call_in_cache method to check cache without calling function (as per note in the code). Add some tests.

cottrell opened the pull request.

cottrell commented:
Code coverage is showing changes in parallel_backends and logger for some reason. No changes there.

pierreglaser commented on joblib/test/test_memory.py:
```diff
@@ -479,6 +479,21 @@ def test_persistence(tmpdir):
 
 
+def test_check_call_in_cache(tmpdir):
+    for func, Result in zip((MemorizedFunc(f, tmpdir.strpath),
```
What is the point of the `Result` object here? I don't see it used.

cottrell replied:
Ah yeah, was just copy/pasting. Cleaned that up now.

cottrell replied:
FYI squashed those into one commit now

pierreglaser commented on joblib/memory.py:
```diff
@@ -782,6 +782,18 @@ def _persist_input(self, duration, args, kwargs, this_duration_limit=0.5):
 
     # XXX: Need a method to check if results are available.
+    def check_call_in_cache(self, *args, **kwargs):
+        """
```
Can you make the docstring follow the [numpy docstring convention](https://numpydoc.readthedocs.io/en/latest/format.html#docstring-standard)? Thanks!

cottrell replied:
Fixed the docstring I think.

cottrell commented:
Just pushed the wrong branch in case someone was about to do something here ... fixing in one sec.

cottrell commented:
> Just pushed the wrong branch in case someone was about to do something here ... fixing in one sec.

Squashed again to one commit. And import removal had crept in and the test had not picked it up. I think the risks are now small as it is only one screen of new lines to view in the single commit.

aabadie commented:
It seems to me that this PR is addressing the same problem as #730. Did you have a look at it first ?

cottrell commented:
> It seems to me that this PR is addressing the same problem as #730. Did you have a look at it first ?

nope. 20 line diff. can't even remember this. Whoever is managing this repo just chuck whatever is easiest. Nobody getting paid here for this so no hard feelings :)  as long and the functionality gets in.

tomMoral commented:
This PR seems to have stalled. I think this would be useful. 
It seems that the only missing parts are an entry in the changelog and a mention in the doc. @cottrell are you available to help getting this merged?

cottrell commented:
@tomMoral just saw this ... I've added something to the doc and the changelog. I'm assuming the pipeline does the auto gen part of the doc stuff.

Rebased on master.

cottrell commented:
Not sure about those test failures, likely the master I rebased on. I don't think those failures happened before the rebase.

danlkv commented:
What happened? Why this is not merged?

tomMoral commented:
I did not notice the update from @cottrell , sorry!
Indeed, we should merge this one. I will rebase on master and try to get this merge asap.

tomMoral commented:
@aabadie @ogrisel I think this feature is helpful and the proposed solution is nice.
Not so sure about the naming.

tomMoral commented on joblib/memory.py:
```diff
@@ -606,6 +609,21 @@ def __getstate__(self):
+
+        Returns
+        -------
+        is_called_in_cache: bool
```
```suggestion
        is_call_in_cache: bool
```

tomMoral commented on joblib/memory.py:
```diff
@@ -606,6 +609,21 @@ def __getstate__(self):
+        Returns
+        -------
+        is_called_in_cache: bool
+            whether or not the function has been cached for the input
+            arguments that have been passed.
```
```suggestion
            Whether or not the result of the function has been cached
            for the input arguments that have been passed.
```

tomMoral reviewed (commented):
Some nitpicks.

tomMoral commented on CHANGES.rst:
```diff
@@ -111,6 +114,7 @@ Release 0.14.1
   `pkg_resources` that introduced a performance regression under Windows:
   https://github.com/joblib/joblib/issues/965
 
+
```
```suggestion```
